pdr_backend/pdr_dashboard/dash_components/plots.py

```diff
@@ -349,33 +349,42 @@ def process_payouts(
     """
     processed = ProcessedPayouts()
 
-    profit = 0.0
-    for p in payouts:
-        processed.predictions += 1
-        processed.tx_cost += tx_fee_cost
-        profit_change = float(max(p["payout"], 0) - p["stake"])
-        profit += profit_change
-        processed.correct_predictions += p["payout"] > 0
-
-        if calculate_confint:
-            acc_l, acc_u = proportion_confint(
-                count=processed.correct_predictions, nobs=processed.predictions
-            )
-
-            processed.acc_intervals.append(
-                AccInterval(
-                    acc_l,
-                    acc_u,
-                )
-            )
+    p = payouts.copy()
+    processed.predictions = len(payouts)
+    p["predictions_crt"] = range(1, len(payouts) + 1)
+    processed.tx_cost = tx_fee_cost * len(payouts)
+    p["profit_change"] = p.apply(
+        lambda x: max(x["payout"], 0) - x["stake"], axis=1
+    ).astype(float)
+    p["correct_prediction"] = p["payout"].apply(lambda x: x > 0).astype(int)
+    p["correct_predictions_crt"] = p["correct_prediction"].cumsum()
+    processed.correct_predictions = p["correct_prediction"].sum()
```

Comment:
For improvements i suggest:
- use a single query to process all this data. Use `with_columns()` to process the payouts once and get results as columns to a 'result' data frame
- use lazy data frames to avoid unnecessary intermediate copies and computations